components/infobox/wikis/chess/infobox_league_custom.lua

```diff
@@ -95,12 +96,23 @@ function CustomLeague:addToLpdb(lpdbData, args)
 	Array.forEach(CustomLeague.getRestrictions(args.restrictions),
 		function(res) lpdbData.extradata['restriction_' .. res.data] = 1 end)
 
+	lpdbData.extradata.modes = args.modes
+
 	return lpdbData
 end
 
 ---@param args table
 function CustomLeague:customParseArguments(args)
-	self.data.mode = MODES[string.lower(self.args.mode or '')] or MODES.classical
+	-- Modes.
+	self.args.modes = {}
```

Comment:
fwiw i think rath would like it more if it were `self.data.modes`

Reply:
Is this possible, I just presumed `data.mode` was a specific field, and there was no `data.modes`?

Reply:
data is just an (abitrary) table we push into self
it is used to pass values to lpdb storage but you certainly can set additional stuff (that doesn't get directly stored in lpdb) into it
the idea was/is to not have to adjust the args table